src/meta-srv/src/service/store/txn.rs

```diff
@@ -0,0 +1,430 @@
+        assert!(txn_response.succeeded);
+
+        let txn = Txn::new()
+            .when(vec![Compare::with_value(
+                key.clone(),
+                CompareOp::Greater,
+                vec![1],
+            )])
+            .and_then(vec![TxnOp::Put(key.clone(), vec![3])])
+            .or_else(vec![TxnOp::Put(key, vec![4])]);
+        let txn_response = kv_store.txn(txn).await.unwrap();
+        assert!(!txn_response.succeeded);
+    }
+
+    #[tokio::test]
+    async fn test_txn_compare_less() {
+        let kv_store = create_kv_store().await;
+        let key = vec![103u8];
+        kv_store.delete(vec![3], false).await.unwrap();
+
+        let txn = Txn::new()
+            .when(vec![Compare::with_not_exist_value(
+                key.clone(),
+                CompareOp::Less,
+            )])
+            .and_then(vec![TxnOp::Put(key.clone(), vec![1])])
+            .or_else(vec![TxnOp::Put(key.clone(), vec![2])]);
+        let txn_response = kv_store.txn(txn.clone()).await.unwrap();
+        assert!(!txn_response.succeeded);
+
+        let txn_response = kv_store.txn(txn).await.unwrap();
+        assert!(!txn_response.succeeded);
+
+        let txn = Txn::new()
+            .when(vec![Compare::with_value(
+                key.clone(),
+                CompareOp::Less,
+                vec![2],
+            )])
+            .and_then(vec![TxnOp::Put(key.clone(), vec![3])])
+            .or_else(vec![TxnOp::Put(key, vec![4])]);
+        let txn_response = kv_store.txn(txn).await.unwrap();
+        assert!(!txn_response.succeeded);
+    }
+
+    #[tokio::test]
+    async fn test_txn_compare_not_equal() {
+        let kv_store = create_kv_store().await;
+        let key = vec![104u8];
+        kv_store.delete(key.clone(), false).await.unwrap();
+
+        let txn = Txn::new()
+            .when(vec![Compare::with_not_exist_value(
+                key.clone(),
+                CompareOp::NotEqual,
+            )])
+            .and_then(vec![TxnOp::Put(key.clone(), vec![1])])
+            .or_else(vec![TxnOp::Put(key.clone(), vec![2])]);
+        let txn_response = kv_store.txn(txn.clone()).await.unwrap();
+        assert!(!txn_response.succeeded);
+
+        let txn_response = kv_store.txn(txn).await.unwrap();
+        assert!(txn_response.succeeded);
+
+        let txn = Txn::new()
+            .when(vec![Compare::with_value(
+                key.clone(),
+                CompareOp::Equal,
+                vec![2],
+            )])
+            .and_then(vec![TxnOp::Put(key.clone(), vec![3])])
+            .or_else(vec![TxnOp::Put(key, vec![4])]);
+        let txn_response = kv_store.txn(txn).await.unwrap();
+        assert!(!txn_response.succeeded);
+    }
+
+    async fn create_kv_store() -> KvStoreRef {
+        std::sync::Arc::new(crate::service::store::memory::MemStore::new())
+        // The same test can be run against etcd by uncommenting the following line
```

Comment:
Can we add a feature to test against etcd in github CI? We can easily establish the etcd service in github action for testing.

Reply:
Perhaps in the next PR, I will leave a TODO.